### ISSUE TITLE
Data nodes crashing in cascade with java.lang.StackOverflowError: null

### ISSUE THREAD

**Issue by fdeschamps** [elastic/elasticsearch#34715]
**Elasticsearch version**: 6.4.2

**Plugins installed**:
xpack.monitoring.enabled: True
xpack.monitoring.collection.enabled: True
xpack.security.enabled: false
xpack.security.audit.enabled: false
node.ml: false
xpack.ml.enabled: false
xpack.watcher.enabled: false

**JVM version**: 
java version "1.8.0_162"
Java(TM) SE Runtime Environment (build 1.8.0_162-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.162-b12, mixed mode)

**OS version**:  4.13.0-41-generic #46~16.04.1-Ubuntu SMP Thu May 3 10:06:43 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

**Description of the problem including expected versus actual behavior**:
Our data nodes are crashing in cascade, several at a time. (Nodes with "node.data: false" don't crash.)

It's on a cluster with 28 datanodes, 4 000 indices, 30 000 shards, 70 To of total data, 90 billions documents and a cluster state of ~100Mo.

It happened 2 times since our upgrade from 5.5.1 seven days ago.

**Steps to reproduce**:
It seems to happens on a search request but I don't know which one.

**Provide logs (if relevant)**:

We can find these logs after the crash :
```
[2018-10-22T16:08:25,404][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [data004] fatal error in thread [elasticsearch[data004][search
][T#7]], exiting
java.lang.StackOverflowError: null
        at org.apache.lucene.util.automaton.DaciukMihovAutomatonBuilder.replaceOrRegister(DaciukMihovAutomatonBuilder.java:322) ~[lucene-core-7.4.0.jar:7.4.0 9060ac689c270b02143f375de0348b7f626adebc - jpountz - 2018-06-18 16:51:45]
        at org.apache.lucene.util.automaton.DaciukMihovAutomatonBuilder.replaceOrRegister(DaciukMihovAutomatonBuilder.java:324) ~[lucene-core-7.4.0.jar:7.4.0 9060ac689c270b02143f375de0348b7f626adebc - jpountz - 2018-06-18 16:51:45]
        at org.apache.lucene.util.automaton.DaciukMihovAutomatonBuilder.replaceOrRegister(DaciukMihovAutomatonBuilder.java:324) ~[lucene-core-7.4.0.jar:7.4.0 9060ac689c270b02143f375de0348b7f626adebc - jpountz - 2018-06-18 16:51:45]
        at org.apache.lucene.util.automaton.DaciukMihovAutomatonBuilder.replaceOrRegister(DaciukMihovAutomatonBuilder.java:324) ~[lucene-core-7.4.0.jar:7.4.0 9060ac689c270b02143f375de0348b7f626adebc - jpountz - 2018-06-18 16:51:45]
        at org.apache.lucene.util.automaton.DaciukMihovAutomatonBuilder.replaceOrRegister(DaciukMihovAutomatonBuilder.java:324) ~[lucene-core-7.4.0.jar:7.4.0 9060ac689c270b02143f375de0348b7f626adebc - jpountz - 2018-06-18 16:51:45]
...
```

The last line repeats itself ~2000 times.

**Comment by albertzaharovits** [comment]
Hello @fdeschamps,

Thank you for reporting.
We would really appreciate the stacktrace in solving this. If you could pin point the query that would be wonderful. I would start the lookout with Regex queries.

CC @elastic/es-search-aggs

**Comment by jimczi** [comment]
Thanks @fdeschamps , this is a duplicate of https://github.com/elastic/elasticsearch/issues/32875, hence closing. The fix is included in the upcoming `6.5` release.

**Comment by fdeschamps** [comment]
Thank you ! I can confirm the search requests leading to the crash were using hightlights.
